src/ec2/model.ts

```diff
@@ -0,0 +1,69 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import * as vscode from 'vscode'
+import { Session } from 'aws-sdk/clients/ssm'
+import { Ec2Selection } from './utils'
+import { getOrInstallCli } from '../shared/utilities/cliUtils'
+import { isCloud9 } from '../shared/extensionUtilities'
+import { withoutShellIntegration } from '../ecs/commands'
+import { ToolkitError } from '../shared/errors'
+import { DefaultSsmClient } from '../shared/clients/ssmClient'
+
+export class Ec2ConnectClient {
```

Comment:
will this be moved to the `.../clients/` folder? If it's meant to be a client, consider naming it `Ec2Client` since it's likely that we'll put all ec2-related client-layer stuff there.

If this is meant to hold higher-layer business logic, could name it simply `ec2` or `ec2Model`.

Reply:
Its meant to hold the logic specific for connecting to remote instance which combines working with the SSM client and EC2 client. I wasn't sure if this qualified as business logic or if it should itself be a higher-level client built on others.

Reply:
That is business logic for EC2. The common parts of "remote connect" could be extracted though (see other comments)